src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        final int datePosition = 0;
+        final int employeePosition = 1;
+        final int hoursPosition = 2;
+        final int ratePosition = 3;
+
+        LocalDate parsedDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate parsedDateTo = LocalDate.parse(dateTo, FORMATTER);
+
+        StringBuilder salaryReport = new StringBuilder("Report for period " + dateFrom + " - "
+                + dateTo + System.lineSeparator());
```

Comment:
Use StringBuilder's `append` method instead of string concatenation when building `salaryReport`. String concatenation inside a loop or similar constructs is less efficient because it creates multiple String objects. StringBuilder is designed for such use cases to improve performance.